### ISSUE TITLE
Flag Extra Properties as Errors / Warnings?

### ISSUE THREAD

**Issue by geoffreywiseman** [codekie/openapi-examples-validator#83]
I've been doing a bit more digging into why `openapi-examples-validator` didn't flag the error I discovered recently.

In this particular case, a field name was updated in the documentation and code but not in the example. The field was optional, so I suspect that OEV simply treated it as unnecessary extra information and a missing optional property, neither of which is strictly an error?

That's exactly how I'd like my API to react if that example is fed to it (e.g. Postel's law), but it's not how I'd want my example validated. Most of the time, my examples are meant to be valid requests, so if there's information in the example that isn't supported by the schema, I probably want to be warned, because it's probably an error in the example.

Is it possible to warn about these, or to have a CLI flag I can turn on to request that I be warned about these?

**Comment by geoffreywiseman** [comment]
Yeah, I can definitely add an undocumented field to an existing example with no new warnings.

**Comment by geoffreywiseman** [comment]
Ah, I see I can turn on `additionalProperties: false`, and then Ajv will flag these ... I'll think about whether or not I should do that. Might be nice to have the option to merge in `additionalProperties: false` into the schema even when it's not in the OAS, but that might also be overkill. Hm.

**Comment by geoffreywiseman** [comment]
Yeah, I'm not the only one who doesn't love the idea of setting `additionalProperties` to false: https://github.com/nutanix/api-guidelines/blob/master/chapters/compatibility.adoc

And yet, I want my examples validated as if ... ;)

**Comment by codekie** [comment]
I see your point. I'll look into it.

**Comment by codekie** [comment]
@geoffreywiseman I have released a new version (`4.0.0-alpha.0`) with the new option:

```
-n, --no-additional-properties             don't allow properties that are not described in the schema
```

The change was a bit more complicated than expected. While I do have automated tests, I'd appreciate any help with "real" examples. Can you please run it over your schema(s) and let me know if it covers your cases?

However, there is one caveat: This new option doesn't work if `allOf` is used in the schema, to merge subschemas.

**Comment by geoffreywiseman** [comment]
Will do -- I've already done the manual version of this by turning on "additionalProperties:false" for each schema and then removing it again, but it won't be hard for me to look at some of things I fixed when I did that and return them to the previous state and validate that OEV catches the same issues I found. Thanks!

**Comment by geoffreywiseman** [comment]
Sorry, finally got back to this. Yes, looks good to me.  It actually found one problem that I thought I had fixed on the last pass but clearly I did not, because here it is being caught again. Made some example changes by changing the names of properties in examples (updating the api but not the example is the most common area where this bites me), and they were immediately picked up.

Not currently using `allOf` although I could see that being a problem for some.

**Comment by codekie** [comment]
Good. Thanks, for your help ☺️ . I will release a new version soon.

**Comment by cbetta** [comment]
Hi, did the issue with `allOf` in combination with `--no-additional-properties` get resolved? I still seem to run into issues where it doesn't resolve the subschema.

**Comment by codekie** [comment]
Hi @cbetta . The schema validator uses json schema draft 4 and there `additionalProperties: false` does not work with `allOf`. See [here](https://stackoverflow.com/questions/22689900/json-schema-allof-with-additionalproperties#answer-23001194)

**Comment by cbetta** [comment]
@codekie interesting. Thank you for the update.
